### PR TITLE
 User-story: 10561 - Do not bypass_reservation_update at _unreserve_inital_demand

### PR DESCRIPTION
**Pull request by ws-mps**
By default it is now bypassed, but in some cases still it shouldn't.

### CODE REVIEW

File: addons/udes_stock/tests/test_move.py [hunk 1]
```diff
@@ -100,3 +100,31 @@ def test03_cancel_backorder_cancels_remaining(self):
...
+
+        self.assertEqual(picking.state, 'assigned')
+        self.assertEqual(apple_quant.reserved_quantity, 4)
+
+        partial_pick_apples = [{'barcode': self.apple.barcode, 'qty': 2}]
+
+        picking.update_picking(product_ids=partial_pick_apples, validate=True, create_backorder=True)
+        self.assertEqual(picking.state, 'done')
+        backorder = Picking.get_pickings(backorder_id=picking.id)
+        self.assertEqual(backorder.state, 'assigned')
```
Comment (apatrick88) [inline addons/udes_stock/tests/test_move.py]: Is it worth asserting the remaining reserved quantity on the backorder picking too?

File: addons/udes_stock/models/stock_move.py [hunk 2]
```diff
@@ -10,25 +10,26 @@
...
 
     def _unreserve_initial_demand(self, new_move):
         """ Override stock default function to keep the old move lines,
             so there is no need to create them again
         """
-        self.mapped('move_line_ids')\
-            .filtered(lambda x: x.qty_done == 0.0)\
-            .write({'move_id': new_move, 'product_uom_qty': 0})
+        self.with_context(bypass_reservation_update=False).mapped("move_line_ids").filtered(
+            lambda x: x.qty_done == 0.0
```
Comment (williamsajr) [inline addons/udes_stock/models/stock_move.py:31]: I don't think this change will work as qty_done is not zero, so this change will not have any effect. We probably want to create a new move line for the remainder on the new move instead anyway

File: addons/udes_stock/tests/test_move.py [hunk 3]
```diff
@@ -100,3 +100,31 @@ def test03_cancel_backorder_cancels_remaining(self):
...
+        Picking = self.env['stock.picking']
+        apple_quant = self.create_quant(self.apple.id, self.test_location_01.id, 4)
+        picking = self.create_picking(self.picking_type_pick, products_info=self.pack_4apples_info, confirm=True, assign=True)
+
+        self.assertEqual(picking.state, 'assigned')
+        self.assertEqual(apple_quant.reserved_quantity, 4)
+
+        partial_pick_apples = [{'barcode': self.apple.barcode, 'qty': 2}]
+
+        picking.update_picking(product_ids=partial_pick_apples, validate=True, create_backorder=True)
```
Comment (williamsajr) [inline addons/udes_stock/tests/test_move.py]: I think we should be updating the move lines (qty_done)

File: addons/udes_stock/tests/test_move.py (hunk 1 above)
Comment (ws-mps) [reply addons/udes_stock/tests/test_move.py]: The problem was it wasn't getting assigned at all so if the state is assigned means we have found stock, so the quantity is not really needed.

**Pull request merged**